opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -73,66 +78,48 @@ def collect(self) -> Optional[Sum]:
         """
         now = _time_ns()
 
-        with self._lock:
-            value = self._value
-            start_time_unix_nano = self._start_time_unix_nano
-
-            self._value = 0
-            self._start_time_unix_nano = now + 1
-
-        return Sum(
-            aggregation_temporality=AggregationTemporality.DELTA,
-            is_monotonic=self._instrument_is_monotonic,
-            start_time_unix_nano=start_time_unix_nano,
-            time_unix_nano=now,
-            value=value,
-        )
+        if self._instrument_temporality is AggregationTemporality.DELTA:
 
+            with self._lock:
+                value = self._value
+                start_time_unix_nano = self._start_time_unix_nano
 
-class AsynchronousSumAggregation(
-    _InstrumentMonotonicityAwareAggregation, Aggregation[Sum]
-):
-    def __init__(self, instrument_is_monotonic: bool):
-        super().__init__(instrument_is_monotonic)
-        self._value = None
-        self._start_time_unix_nano = _time_ns()
+                self._value = 0
+                self._start_time_unix_nano = now + 1
 
-    def aggregate(self, measurement: Measurement) -> None:
-        with self._lock:
-            self._value = measurement.value
+            return Sum(
+                aggregation_temporality=AggregationTemporality.DELTA,
+                is_monotonic=self._instrument_is_monotonic,
+                start_time_unix_nano=start_time_unix_nano,
+                time_unix_nano=now,
+                value=value,
+            )
 
-    def collect(self) -> Optional[Sum]:
-        """
-        Atomically return a point for the current value of the metric.
-        """
         if self._value is None:
             return None
 
         return Sum(
-            start_time_unix_nano=self._start_time_unix_nano,
-            time_unix_nano=_time_ns(),
-            value=self._value,
             aggregation_temporality=AggregationTemporality.CUMULATIVE,
             is_monotonic=self._instrument_is_monotonic,
+            start_time_unix_nano=self._start_time_unix_nano,
+            time_unix_nano=now,
+            value=self._value,
         )
 
 
 class LastValueAggregation(Aggregation[Gauge]):
     def __init__(self):
         super().__init__()
-        self._value = None
+        self._value = 0
 
     def aggregate(self, measurement: Measurement):
         with self._lock:
             self._value = measurement.value
 
-    def collect(self) -> Optional[Gauge]:
+    def collect(self) -> Gauge:
         """
         Atomically return a point for the current value of the metric.
         """
-        if self._value is None:
-            return None
-
```

Comment:
I think the changes to `LastValueAggregation` should be undone. If you set `LastValueAggregation` on a sync or async instrument, you wouldn't want to send a new point if there were no measurements in a given collection interval.

Reply:
right, undone :+1: